python/pyarrow/scalar.pxi

```diff
@@ -838,6 +843,67 @@ cdef class UnionScalar(Scalar):
         return sp.type_code
 
 
+cdef class ExtensionScalar(Scalar):
+    """
+    Concrete class for Extension scalars.
+    """
+
+    @property
+    def value(self):
+        """
+        Return storage value as a scalar.
+        """
+        cdef CExtensionScalar* sp = <CExtensionScalar*> self.wrapped.get()
+        return Scalar.wrap(sp.value) if sp.is_valid else None
```

Comment:
This could also be a null scalar of the storage type?

Reply:
Returning `None` mirrors the behaviour on the C++ side, though.